Scripts/Weapon/DashSkill.cs

```diff
@@ -0,0 +1,245 @@
+        damage = 30.0f;
+        impactRange = 5.0f;
+        coolAmount = 0.4f;
+    }
+
+    public override void _Ready()
+    {
+        base._Ready();
+        player = GetParent<Player>();
+        overheatingGun = player.GetNode<OverheatingGun>("Weapon");
+
+        (AimIndicator as GeneralAimIndicator).IndicatorType = GeneralAimIndicator.AimIndicatorType.CONICAL;
+        (AimIndicator as GeneralAimIndicator).SpreadDegrees = spreadDegrees;
+        targetArea = new Area();
+        targetArea.AddChild(new CollisionShape() { Shape = new CylinderShape { Radius = range } });
+
+        coneSides = new Area();
+        coneSides.AddChild(new CollisionShape
+        {
+            Shape = new BoxShape { Extents = new Vector3(0.01f, 1, range / 2) },
+            RotationDegrees = Vector3.Up * spreadDegrees / 2,
+            Translation = range / 2 * Vector3.Forward.Rotated(Vector3.Up, Mathf.Deg2Rad(spreadDegrees / 2))
+        });
+        coneSides.AddChild(new CollisionShape
+        {
+            Shape = new BoxShape { Extents = new Vector3(0.01f, 1, range / 2) },
+            RotationDegrees = Vector3.Up * -spreadDegrees / 2,
+            Translation = range / 2 * Vector3.Forward.Rotated(Vector3.Up, Mathf.Deg2Rad(-spreadDegrees / 2))
+        });
+
+        impactArea = new Area();
+        impactArea.AddChild(new CollisionShape() { Shape = new CylinderShape { Radius = impactRange } });
+
+        hintReticle = new Spatial();
+        hintReticle.AddChild(new CSGBox
+        {
+            Scale = 0.5f * Vector3.One,
+            Translation = 2 * Vector3.Up,
+            Material = new SpatialMaterial
+            {
+                AlbedoColor = Colors.Green
+            }
+        });
+        impactAimIndicator = new ConicalAimIndicator { SpreadDegrees = 360, Range = impactRange };
+        hintReticle.AddChild(impactAimIndicator);
+
+        AddChild(targetArea);
+        AddChild(coneSides);
+        AddChild(impactArea);
+        AddChild(hintReticle);
+    }
+
+    public override void _PhysicsProcess(float delta)
+    {
+        if (isRunning)
+            return;
+        if (showTargetHint)
+        {
+            targetEnemy = GetTargetEnemy();
+            hintReticle.Visible = targetEnemy != null;
+
+            if (targetEnemy != null)
+            {
+
+                Vector3 enemyLocation = targetEnemy.GlobalTransform.origin;
+                Dictionary ray = GetWorld().DirectSpaceState.IntersectRay(GlobalTransform.origin, enemyLocation, collisionMask: 4);
+
+                if (ray.Count == 0)
+                    return;
+                Vector3 enemyBoundary = (Vector3)ray["position"];
+
+                impactAimIndicator.Translation = enemyBoundary - enemyLocation;
+
+                Transform hintReticleGlobalTransform = hintReticle.GlobalTransform;
+                hintReticleGlobalTransform.origin = enemyLocation;
+                hintReticle.GlobalTransform = hintReticleGlobalTransform;

```

Comment:
Can you do `hintReticle.GlobalTransform.origin = enemyLocation`?

Reply:
cannot, because GlobalTransform is a property, not a variable (it's only made in the getter and returned)